src/isql/isql.epp

```diff
@@ -329,6 +329,11 @@ static int win32ReadConsole(FILE* file, char* buffer, size_t bufferSize)
 		return -1;
 	}
 
+	const size_t MAX_LINE_LENGTH = 32765;
+	static WCHAR wideBuf[MAX_LINE_LENGTH +1];
```

Comment:
Curiosity question: should this be allocated on the heap instead of on the stack? 

For example Visual Studio analysis will issue a C6262 warning for this, and its caller (readNextInputLine) also allocates a buffer of 64 KiB on the stack, so these two methods combined will allocate 128 KiB+ on the stack.

Reply:
> Curiosity question: should this be allocated on the heap instead of on the stack?
> 
> For example Visual Studio analysis will issue a C6262 warning for this, and its caller (readNextInputLine) also allocates a buffer of 64 KiB on the stack, so these two methods combined will allocate 128 KiB+ on the stack.

This is `static` buffer, it is not on stack. And it is done intentionally.

Reply:
OK, I had missed that, but the similar buffer in `readNextInputLine` is not static. Should it be?

Reply:
`pWide` and `charsRead` are more obvious candidates for static duration. Or at least a comment why they are fine as local variables.

Reply:
> OK, I had missed that, but the similar buffer in `readNextInputLine` is not static. Should it be?

It is not necessary.